cgo_go122.go

```diff
@@ -0,0 +1,30 @@
+//go:build go1.22
+
+package tokenizers
+
+/*
+// This is a performance optimization.
+// The following noescape and nocallback directives are used to
+// prevent the Go compiler from allocating function parameters on the heap.
```

Comment:
did you notice changes in bencmarks?

Reply:
Just prevent some parameters allocated on the heap. 😄